src/ucp/core/ucp_ep.c

```diff
@@ -74,6 +74,17 @@ typedef struct ucp_ep_discard_lanes_arg {
 } ucp_ep_discard_lanes_arg_t;
 
 
+/**
+ * Failed UCT EP placeholder
+ */
+typedef struct {
+    uct_ep_t     super;
+    ucs_status_t status;
```

Comment:
it seems like shared status across many EPs also can be wrong

Reply:
the first failed lane should determine the error code on the whole ucp ep

Reply:
But as far as I can see it's static instance which is shared across all UCP EPs per process

Reply:
in case of discard, there is another instance that is dynamically-allocated